gobblin-runtime/src/main/java/gobblin/runtime/locks/JobLock.java

```diff
@@ -27,40 +28,51 @@
  *
  * @author Yinan Li
  */
-public interface JobLock {
+public interface JobLock extends Closeable {
+
+  /**
+   * Initializes the lock.
+   *
+   * @param properties  the job properties
+   * @param jobLockEventListener the listener for lock events
+   * @throws JobLockException thrown if the {@link JobLock} fails to initialize
+   */
+  void initialize(Properties properties, JobLockEventListener jobLockEventListener)
```

Comment:
Can we have an `abstract` class called `ListenableJobLock` that takes in a `Properties` and a `JobLockEventListener` in the constructor. Would rather avoid adding an `initialize` method to the interface as it is a backwards incompatible change to the API.